include/revng/Yield/Assembly/DisassemblyHelper.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <map>
+#include <memory>
+
+#include "revng/Yield/Assembly/Assembly.h"
+
+namespace detail {
+class HiddenDisassemberStorage;
+}
+namespace model {
+class Function;
+}
+namespace efa {
+class FunctionMetadata;
+}
+class DisassemblerInstance;
+class RawBinaryView;
+
+class DissassemblyHelper {
+private:
+  detail::HiddenDisassemberStorage *Internal;
```

Comment:
Why pimpl? Caliban was doing this a lot, but I'm not really a fan of it.
Consider dropping the pimpl pattern.

Reply:
I hate the idea of having to include all that ~~sketchy~~ internal LLVM stuff.

I'd prefer that as few of my translation units include `MCDisassembler/MCDisassembler.h` and friends. Even if it means pimpl.

Reply:
I tried everything I could to make either one of those work. But I kept getting `sizeof` problem for some reason - in the end I just gave up and went with the ugly but simple approach.

Reply:
If I recall correctly, it boiled down to the fact that every single translation unit including the problem header also had to include the original llvm stuff - otherwise it wouldn't compile properly. Which kind of defeats the purpose of forward declaring this stuff.

Reply:
On godbolt I was having `sizeof` issues if I left the inline constructor, adding an out-of-line constructor fixed the problem for me.

Can you get to the bottom of this? It's good to know for the future too.

Reply:
Wait, this actually be relevant. I'll give it a try.

Reply:
Seems to work: https://github.com/revng/revng/pull/173/commits/c0975737d3057076daadb9c08ed0c007a4dfc479.

Also I went with pimpl after all. Now that we disclosed internals of the `interface` as private methods - it does rely on those headers after all.

But, tbh, I like it this way a bit more - let the `interface` be a private-ish header that should be only touched when it's needed while the `helper` is the official interface.

Reply:
OK.
Maybe rename `detail::HiddenDisassemberStorage` `DissassemblyHelperImpl`.

Reply:
Done: https://github.com/revng/revng/pull/173/commits/b2f9dbc3c7141222fa5b7c16eab9c996004fd000.